src/Microsoft.Health.Fhir.Core/Models/ResourceChangeData.cs

```diff
@@ -0,0 +1,24 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Fhir.Core.Models
+{
+    public class ResourceChangeData : IResourceChangeData
```

Comment:
>IResourceChangeData [](http://example.com/codeflow?start=38&length=19)

What do you get out of having this interface?
What purpose does it serves?

Reply:
It is general practice to use abstraction to enable various implementations for different purposes such as testing.

Reply:
In general, yes, interfaces are very powerful tool for decoupling.
But in this particular case, does it bring anything?

Reply:
Basically defining a contract for the resource change data. So, the published event worker can define an input contract using the interface. Also, it can be extended later.

Reply:
This PR has class `SqlServerFhirResourceChangeDataStore` which derived from `IChangeFeedSource<T>`

`IChangeFeedSource` is your main contract of getting data. Why `T` has to be interface. What do you get out of this second degree of freedom? 

I could understand some logic if interface contains methods, but that interface is just plain POCO class and maps to a POCO class. 

Can you provide example how you plan to use it in future as interface, because in current PR there is no need for that interface.

Reply:
Removed the IResourceChangeData interface due to no functionality and it is only used for a POCO class. Will add later if needed.